include/rocksdb/advanced_options.h

```diff
@@ -655,11 +655,15 @@ struct AdvancedColumnFamilyOptions {
   // unit: seconds. Ex: 1 day = 1 * 24 * 60 * 60
   // In FIFO, this option will have the same meaning as
   // periodic_compaction_seconds. Whichever stricter will be used.
+  // 0 means disabling.
+  // UINT64_MAX - 1 (0xfffffffffffffffe) is special flag to allow RocksDB to
```

Comment:
why not just use UINT64_MAX?
Are you concerned that it might collide somewhere with the default for `periodic_compaction_seconds`?

Reply:
I'm concerned that people might already use UINT64_MAX to indicate infinite. It's less likely to use UINT64_MAX - 1.

Reply:
I don't think so. But I don't have a strong preference for either of these values (i.e. UINT64_MAX vs UINT64_MAX - 1), so I'll let you decide the best value.

If that is indeed the case may be it would be good to set the default value of `periodic_compaction_seconds` also to be this value, just to match.

Reply:
Actually I tried and oddly most tests failed. No idea why so I gave it up for now. Will give it another try when I have time.

Reply:
Should we change the default for `periodic_compaction_seconds` or leave it as is?

Reply:
I think I might know the issue, as I ran into something similar. You might need to change some code in version_set.cc similar to PR #5865 .

Reply:
Hmm.. This is odd. It means in somewhere of the code, the option came to there without being sanitized?

Reply:
I think I might have figured out the reason. Will send out a separate PR.

Reply:
Let me do it in the same PR as there will be conflicts.

Reply:
Still some test faiures...